lib/stripe/core_resources/payment_intent.ex

```diff
@@ -0,0 +1,294 @@
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+  require Stripe.Util
+
+  @type last_payment_error :: %{
+          type: String.t(),
+          charge: String.t(),
+          code: String.t(),
+          decline_code: String.t(),
+          doc_url: String.t(),
+          message: String.t(),
+          param: String.t(),
+          source: Stripe.Card.t() | map
+        }
+
+  @type next_action :: %{
+          redirect_to_url: redirect_to_url | nil,
+          type: String.t(),
+          use_stripe_sdk: map | nil
+        }
+
+  @type redirect_to_url :: %{
+          return_url: String.t(),
+          url: String.t()
+        }
+
+  @type transfer_data :: %{
+          :destination => String.t()
+        }
+
+  @type t :: %__MODULE__{
+          id: Stripe.id(),
+          object: String.t(),
+          amount: non_neg_integer,
+          amount_capturable: non_neg_integer,
+          amount_received: non_neg_integer,
+          application: Stripe.id() | nil,
+          application_fee_amount: non_neg_integer | nil,
+          canceled_at: Stripe.timestamp() | nil,
+          cancellation_reason: String.t() | nil,
+          capture_method: String.t(),
+          charges: Stripe.List.t(Stripe.Charge.t()),
+          client_secret: String.t(),
+          confirmation_method: String.t(),
+          created: Stripe.timestamp(),
+          currency: String.t(),
+          customer: Stripe.id() | Stripe.Customer.t() | nil,
+          description: String.t() | nil,
+          last_payment_error: last_payment_error | nil,
+          livemode: boolean,
+          metadata: Stripe.Types.metadata(),
+          next_action: next_action | nil,
+          on_behalf_of: Stripe.id() | Stripe.Account.t() | nil,
+          payment_method_types: list(String.t()),
+          receipt_email: String.t() | nil,
+          review: Stripe.id() | Stripe.Review.t() | nil,
+          shipping: Stripe.Types.shipping() | nil,
+          source: Stripe.Card.t() | map,
+          statement_descriptor: String.t() | nil,
+          status: String.t(),
+          transfer_data: transfer_data | nil,
+          transfer_group: String.t() | nil
+        }
+
+  defstruct [
+    :id,
+    :object,
+    :amount,
+    :amount_capturable,
+    :amount_received,
+    :application,
+    :application_fee_amount,
+    :canceled_at,
+    :cancellation_reason,
+    :capture_method,
+    :charges,
+    :client_secret,
+    :confirmation_method,
+    :created,
+    :currency,
+    :customer,
+    :description,
+    :last_payment_error,
+    :livemode,
+    :metadata,
+    :next_action,
+    :on_behalf_of,
+    :payment_method_types,
+    :receipt_email,
+    :review,
+    :shipping,
+    :source,
+    :statement_descriptor,
+    :status,
+    :transfer_data,
+    :transfer_group
+  ]
+
+  @plural_endpoint "payment_intents"
+
+  @doc """
+  Create a payment intent.
+   See the [Stripe docs](https://stripe.com/docs/api#create_charge).
```

Comment:
Mind updating the doc block with this link?

https://stripe.com/docs/api/payment_intents/create